packages/files/src/main.rs

```diff
@@ -93,8 +125,31 @@ fn transform_image(input: Value, to: &str) {
             match fs::read(path) {
                 Ok(contents) => {
                     let encoded: String = general_purpose::STANDARD_NO_PAD.encode(contents);
-                    let html = format!("<img src=\"data:image/png;base64, {encoded} \"/>");
-                    let json = json!({"name": "raw", "data": html}).to_string();
+                    let width = input["arguments"]["width"].as_f64().unwrap();
+                    let caption = input["arguments"]["caption"].as_str().unwrap();
+                    let label = input["arguments"]["label"].as_str().unwrap();
+
+                    let id = if label.is_empty() {
+                        String::new()
+                    } else {
+                        format!("id=\"{label}\"")
+                    };
+
+                    let percentage = (width * 100.0).round() as i32;
+                    let style = format!("style=\"width:{percentage}%\"");
+
+                    let mut v = vec![];
+
+                    v.push(String::from("<figure>"));
```

Comment:
Maybe it's a nice idea to only include the figure element if a caption is provided?

Reply:
I'm aware I marked this as resolved earlier, but I've realized the `<figure>` tag slightly resizes the image inside. To me it doesn't seem very intuitive for the user that it gets resized because of a caption. Thoughts?

Reply:
I think it is nice to have it being a figure anyways. This solves one problem I found very annoying; if you have an 4k image (or another large image, which you may get from a phone or something) then the whole image would be shown in the source size (which is huge, you can impossibly see the whole image at once but just one corner). Additionally, it was very dependent on screen resolution. The addition of `<figure>` makes it (relative to) the width of the page in both HTML and LaTeX, which I find consistent and good; I think this is more like what an user expects than for one corner of the image to be shown.

Actually, I chimed in to try out this very thing; since I assumed the previous behavior was present, I was curious how the size argument would work, like if you made it `1.0` it would be great for LaTeX but bad for HTML and vice versa if you set it to the likes of `0.2`. But I clearly see that my worries are solved with the figure tag, so I think that is great! The only problem is that it upscales small images, but I think it is fine anyways.

Good job with the module overall!